alf/algorithms/multi_step_td_loss.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) 2019 Horizon Robotics. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gin
+import torch
+import torch.nn as nn
+
+import alf
+from alf.data_structures import TrainingInfo, LossInfo, StepType
+from alf.utils.losses import element_wise_squared_loss
+from alf.utils import tensor_utils, value_ops
+from alf.utils.summary_utils import safe_mean_hist_summary
+
+
+@gin.configurable
+class MultiStepTDLoss(nn.Module):
```

Comment:
It seems that this loss has quite some overlap with ActorCriticLoss?

Reply:
I am aware of it. I choose not to extract common part to make things clean.